File: api/contextMenu/definitions/contextMenu.ts
export interface ContextMenuElement {
    title: string;
    description?: string;
    /** Full font awesome 5 icon including prefix such as "fas fa-" */
    icon?: string;
    disabled?: boolean;
    onSelect?: () => void;
}

export interface ContextMenuOptions {
    id: string;
    title: string;
    elements: ContextMenuElement[];
    /** Gets called when the menu is closed by the user or using closeMenu as well as when an element is selected */
    onClose?: () => void;
}

export interface ClientContextMenuProvider {
    openMenu(options: ContextMenuOptions): void;
    closeMenu(): void;
}

File: api/contextMenu/definitions/contextMenu.lua
---@class ContextMenuElement
---@field title string
---@field description? string
--- Full font awesome 5 icon including prefix such as "fas fa-"
---@field icon? string
---@field disabled? boolean
---@field onSelect? fun(): nil

---@class ContextMenuOptions
---@field id string
---@field title string
---@field elements ContextMenuElement[]
---@field onClose? fun(): nil

---@class ClientContextMenuProvider
---@field openMenu fun(options: ContextMenuOptions): nil
---@field closeMenu fun(): nil
